### PR TITLE
thingy53: samples and applications: Adding SMP reassembly to Thingy:53 samples

### PR DESCRIPTION
**Pull request by philips77**
Adding SMP reassembly to all samples for Thingy:53. The changes are based on the SMP Server sample.

Also, the PR increases version for those samples to `2.0.0+0`.

Signed-off-by: Aleksander Nowakowski <Aleksander.Nowakowski@nordicsemi.no>

### CODE REVIEW

File: applications/machine_learning/configuration/thingy53_nrf5340_cpuapp/prj.conf [hunk 1]
```diff
@@ -130,6 +130,14 @@ CONFIG_MCUMGR_SMP_BT_CONN_PARAM_CONTROL=y
 CONFIG_MCUMGR_CMD_IMG_MGMT=y
 CONFIG_MCUMGR_CMD_OS_MGMT=y
 
+# Enable the mcumgr Packet Reassembly feature over Bluetooth and its
```
Comment (de-nordic) [inline applications/machine_learning/configuration/thingy53_nrf5340_cpuapp/prj.conf:133]: `MCUmgr`, at least that is how it is spelled in next line; the `the` does not seem to be needed for it either.

Comment (de-nordic) [reply applications/machine_learning/configuration/thingy53_nrf5340_cpuapp/prj.conf]: > Fixed. I'm not sure about the `the`... let me ask around.

I have checked in "My Grammar & I" and assuming that MCUmgr is proper name of a library, then it does not require "the" (p.41, "No need for introductions...").

File: applications/machine_learning/configuration/thingy53_nrf5340_cpuapp/prj.conf [hunk 2]
```diff
@@ -80,7 +80,7 @@ CONFIG_EI_WRAPPER_DATA_BUF_SIZE=1000
```
Comment (MarekPieta) [inline applications/machine_learning/configuration/thingy53_nrf5340_cpuapp/prj.conf:80]: Please add complete commit message (refer to commits merged to main for reference) and squash the commits before merging.

**Review by MarekPieta — APPROVED** [review]

**Comment by kkasperczyk-no** [comment]
@philips77 what is the reason of adding this SMP reassembly feature? Is it something related only to the Thingy:53 or rather to the MCUmgr itself, so the same change should be introduced for all other samples and other boards using SMP?

**Review by kapi-no — APPROVED** [review]

**Comment by de-nordic** [comment]
> @philips77 what is the reason of adding this SMP reassembly feature? Is it something related only to the Thingy:53 or rather to the MCUmgr itself, so the same change should be introduced for all other samples and other boards using SMP?

The SMP reassembly feature allows to have MTU smaller than mcumgr buffer and mcumgr will then re-assembly SMP packet from MTU fragments. The feature is optional because it uses a little bit more flash and is not always needed, for example when your transport is able to handle mtu of size of mcumgr buffer re-assembly is not needed at all.

**Review by de-nordic — APPROVED** [review]

**Pull request merged**